src/Compilation.zig

```diff
@@ -393,7 +393,8 @@ pub const AllErrors = struct {
             indent: usize,
         ) anyerror!void {
             const stderr = stderr_file.writer();
-            const counting_stderr = std.io.countingWriter(stderr).writer();
+            var counting_writer = std.io.countingWriter(stderr);
+            const counting_stderr = counting_writer.writer();
```

Comment:
Interesting. I did not really encounter any problem with this but yes, when I wrote it I was wondering whether it would compile because it's using a temporary. Shouldn't this have emitted a compile error if it wasn't valid? Did you encounter any UB I didn't notice?